### PR TITLE
Improve discoverability of structured search

### LINKED ISSUE #1126: Document query syntax in UI
It's currently not possible to discover that the results can be filtered in the UI.

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
Adds some autocomplete behaviour for the search for basic atoms.
Adds a small link to some docs about the search syntax. Adds the docs, too.

Fixes #1126

### CODE REVIEW

**Comment by lukebjerring** [comment]
Note: the link is broken until this lands, but if you replace `.../blob/master/...` with `.../blob/2641ea94ee4afc6947d05911bc7ae96ab098c428/...` it will show what it'll look at.

File: api/query/README.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
+# wpt.fyi Search queries
```
Comment (foolip) [inline api/query/README.md]: Great docs!

File: api/query/README.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+Combines filters, such that any must apply, e.g.
+
+    chrome:pass or chrome:ok
+
+> NOTE: Or-conjuction takes less precedence than `and`. Precedence can be modified
+> using parens, e.g. `chrome:pass and (firefox:!pass or safari:!pass)`
+
+### Sequential
+
+    seq([query1] [query2] [...])
```
Comment (foolip) [inline api/query/README.md]: I already saw this live, but neat!

File: api/query/README.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+> NOTE: If, rather than a specific set of runs, the user wishes to query for the latest
+> results for a set of products, the `/api/search` endpoint supports the same query
+> parameters as /api/runs, outlined [in the API docs](../README.md)
+
+### Structured query objects
+
+Structured query objects are produced by the syntax parser on wpt.fyi.
+
+The easiest way to build the query you need is to use the syntax above, and inspect
```
Comment (foolip) [inline api/query/README.md]: Would it be hard to take the human-readable query in the API as well?

File: webapp/components/test-search.js [hunk 4]
```diff
@@ -9,8 +9,28 @@ import { html } from '../node_modules/@polymer/polymer/lib/utils/html-tag.js';
...
+  'crash',
+  'skip',
+  'assert',
+  'unknown',
+];
+
+const atoms = {
+  status: statuses,
+};
+for (const b of DefaultBrowserNames) {
```
Comment (foolip) [inline webapp/components/test-search.js]: Another blank line here?
```suggestion

for (const b of DefaultBrowserNames) {
```

File: webapp/components/test-search.js [hunk 5]
```diff
@@ -184,12 +192,19 @@ class TestSearch extends WPTFlags(PolymerElement) {
         padding: 0.5em 0;
         width: 100%;
       }
+      .help {
+        font-size: x-small;
```
Comment (foolip) [inline webapp/components/test-search.js]: The computed size for me is 10px. I suspect this is smaller than would get past an accessibility review, if we ever do that.

File: api/query/README.md (hunk 3 above)
Comment (lukebjerring) [reply api/query/README.md]: Yes, since there's no ohm equivalent for Golang.

File: webapp/components/test-search.js (hunk 5 above)
Comment (lukebjerring) [reply webapp/components/test-search.js]: I intend to deprecate this with a search builder in the future.

**Pull request merged**